Realm/Tests/QueryTests.m

```diff
@@ -1748,6 +1749,25 @@ - (void)testLinksToDeletedOrMovedObject
     XCTAssertEqualObjects(@"Spot's owner", [spotQuery.firstObject name]);
 }
 
+- (void)testSubqueries
+{
+    RLMRealm *realm = [RLMRealm defaultRealm];
+
+    [realm beginWriteTransaction];
+    [CompanyObject createInRealm:realm
+                       withValue:@[@"first company", @[@{@"name": @"John", @"age": @30, @"hired": @NO},
+                                                       @{@"name": @"Jill",  @"age": @40, @"hired": @YES},
+                                                       @{@"name": @"Joe",  @"age": @40, @"hired": @YES}]]];
+    [CompanyObject createInRealm:realm
+                       withValue:@[@"second company", @[@{@"name": @"Bill", @"age": @35, @"hired": @YES},
+                                                        @{@"name": @"Don",  @"age": @45, @"hired": @NO},
+                                                        @{@"name": @"Tim",  @"age": @60, @"hired": @NO}]]];
+    [realm commitWriteTransaction];
+
+    XCTAssertEqual(1U, [CompanyObject objectsWhere:@"SUBQUERY(employees, $employee, $employee.age > 30 AND $employee.hired = FALSE).@count > 0"].count);
+    XCTAssertEqual(2U, [CompanyObject objectsWhere:@"SUBQUERY(employees, $employee, $employee.age < 30 AND $employee.hired = TRUE).@count == 0"].count);
```

Comment:
Would be good to use RLMAssertCount here to at least minimally test handover of subqueries.